projects/components/src/splitter/splitter.component.ts

```diff
@@ -87,213 +95,165 @@ export class SplitterComponent implements OnChanges, AfterContentInit {
     private readonly renderer: Renderer2
   ) {}
 
-  public ngAfterContentInit(): void {
-    this.contents$ = queryListAndChanges$(this.contents ?? EMPTY).pipe(map(contents => contents.toArray()));
-  }
-
   public ngOnChanges(changes: TypedSimpleChanges<this>): void {
     if (changes.splitterSize || changes.direction) {
       this.setSplitterSizeStyle();
+      this.classes = this.buildClasses();
     }
   }
 
-  protected onGutterMouseDown(event: MouseEvent, index: number) {
-    this.resizeStart(event, index);
-    this.bindMouseListeners();
+  public ngAfterContentInit(): void {
+    this.subscribeToQueryListChanges();
   }
 
-  protected readonly getFlex = (dimension: SplitterCellDimension): string => {
-    if (dimension.unit === 'PX') {
-      return `1 1 ${dimension.value}${dimension.unit.toLowerCase()}`;
-    } else {
-      return `${dimension.value} ${dimension.value} 0`;
-    }
-  };
-
-  protected readonly getMaxHeight = (dimension: SplitterCellDimension): string | undefined => {
-    if (dimension.unit === 'PX') {
-      return `${dimension.value}${dimension.unit.toLowerCase()}`;
-    }
-  };
+  private buildNormalizationParams(contents: SplitterContentDirective[]): NormalizationParameters {
+    const totalAvailablePx = this.getElementSizePx(this.element.nativeElement);
+    const splitterPx = this.splitterSize * (contents.length - 1);
+    const pxPerItem = (totalAvailablePx - splitterPx) / contents.length;
 
-  protected horizontal() {
-    return this.direction === SplitterDirection.Horizontal;
+    return {
+      itemCount: contents.length,
+      pxPerItem: pxPerItem
+    };
   }
 
-  private setSplitterSizeStyle(): void {
-    if (this.direction === SplitterDirection.Vertical) {
-      this.splitterSizeStyle = {
-        height: `${this.splitterSize}px`
-      };
-    }
+  private normalizeContentsDimensions(contents: SplitterContentDirective[]): SplitterContentDirective[] {
+    this.normalizationParameters = this.buildNormalizationParams(contents);
 
-    if (this.direction === SplitterDirection.Horizontal) {
-      this.splitterSizeStyle = {
-        width: `${this.splitterSize}px`
-      };
-    }
+    return contents.map(content => this.normalizeContentDimension(content));
   }
 
-  private bindMouseListeners() {
-    if (!this.mouseMoveListener) {
-      this.mouseMoveListener = this.renderer.listen(this.document, 'mousemove', event => {
-        this.debouncedOnResize(event);
-      });
+  private normalizeContentDimension(content: SplitterContentDirective): SplitterContentDirective {
+    if (content.dimension.unit === 'FR') {
+      content.dimension = {
+        unit: 'PX',
```

Comment:
could you use an Enum instead of a string for PX and FR?

Reply:
Yeah, we could. I could have created one, but just left it as a string since that's what it was originally.